### PR TITLE
Fail to upload package when description rendering is invalid.

### LINKED ISSUE #3966: Fail upload if rST description does not render
If the reStructuredText `long_description` does not render, let's fail the upload instead of just rendering it as plaintext, and raise a meaningful error to the user.

### PR DESCRIPTION
**Pull request by mblayman**
This branch prevents a package upload if the rendered content for the
description_content_type is invalid. If no description_content_type is
set, 'text/x-rst' is assumed as the type.

I'm sure the error message could point to something more informative that explains why failed rendering blocks an upload. Since such a page doesn't exist for now, I went with a somewhat generic error message. I'm happy to push more commits when a specific thing is in place. I'm a bit ignorant about how docs are handled for warehouse.

Fixes #3966

### CODE REVIEW

File: warehouse/forklift/legacy.py [hunk 1]
```diff
@@ -940,6 +940,26 @@ def file_upload(request):
...
+            use_fallback=False)
+        if rendered is None:
+            raise _exc_with_message(
+                HTTPBadRequest,
+                ("The description failed to render "
+                 "for '{description_content_type}'. "
+                 "See {projecthelp} "
+                 "for more information.").format(
+                    description_content_type=description_content_type,
+                    projecthelp=request.help_url(_anchor='project-name'),
```
[the remark this answers is not in the document]
Comment (mblayman) [reply warehouse/forklift/legacy.py]: Is there any particular person who deals with docs specifically that I should ping or should I take a crack at the documentation myself?

File: warehouse/forklift/legacy.py [hunk 2]
```diff
@@ -940,6 +940,26 @@ def file_upload(request):
...
+        if not description_content_type:
+            description_content_type = 'text/x-rst'
+        rendered = readme.render(
+            form.description.data, description_content_type,
+            use_fallback=False)
+        if rendered is None:
+            raise _exc_with_message(
+                HTTPBadRequest,
+                ("The description failed to render "
+                 "for '{description_content_type}'. "
```
[the remark this answers is not in the document]
Comment (mblayman) [reply warehouse/forklift/legacy.py]: Sure thing. I can do that.

File: warehouse/forklift/legacy.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (di) [reply warehouse/forklift/legacy.py]: Nope! Feel free to do this yourself in this PR.

**Comment by mblayman** [comment]
@di Please let me know if you need any more for the docs or the error message. I did my best, however, PyPI might have some guidelines or voice that I'm unaware of, and I can tweak the wording until it feels right.

**Comment by mblayman** [comment]
Woohoo! Thanks for the help @di! I'm glad I was able to fix a PyPI issue that has bugged me for ages.

**Pull request merged**